storage/innobase/xtrabackup/src/changed_page_tracking.cc

```diff
@@ -0,0 +1,303 @@
+/******************************************************
+* Copyright (c) 2021 Percona LLC and/or its affiliates.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+
+*******************************************************/
+
+/* Changed page tracking implementation */
+#include "changed_page_tracking.h"
+#include <iostream>
+#include "backup_mysql.h"
+#include "common.h"
+#include "components/mysqlbackup/backup_comp_constants.h"
+#include "srv0srv.h"
+#include "xtrabackup.h"
+
+namespace pagetracking {
+
+/** @return short form of UUID
+@param[in] conn connection handle to MySQL server */
+static unsigned long long int get_uuid_short(MYSQL *connection) {
+  char *uuid = read_mysql_one_value(connection, "SELECT UUID_SHORT()");
+  unsigned long long uuid_short = strtoull(uuid, nullptr, 10);
+  free(uuid);
+  return (uuid_short);
+}
+
+/** Read the disk page tracking file and build the changed page tracking map for
+the LSN interval incremental_lsn to checkpoint_lsn_start.
+@param[in] checkpoint_lsn_start  start checkpoint lsn
+@param[in] connection            MySQL connectionn
+@return the built map or nullptr if unable to build for any reason. */
+xb_space_map *init(lsn_t checkpoint_lsn_start, MYSQL *connection)
+{
+  const ulong page_tracking_read_buffer_size = ((srv_log_buffer_size) / 2);
```

Comment:
constexpr?

Reply:
srv_log_buffer_size is not const so complier complaints. 
read of non-const variable 'srv_log_buffer_size' is not allowed in a constant expression